test/api.js

```diff
@@ -238,4 +238,62 @@ tape('[Transaction]: Basic functions', function (t) {
     st.equal(tx.getChainId(), 0x16b2)
     st.end()
   })
+
+  t.test('automatically prepend 0x to transaction fields when data is passed as object', function (st) {
```

Comment:
I think we should also add some negative tests to make sure that it doesn't prepend `0x` for non-string. Easy to catch now, but it's easy to add a few negative tests to ensure minimal regression in the future.

Reply:
Current tests should cover this as they simply check that txes with properties of equal value (but different types and differences in `0x` prefix) produce equal hashes and serializations. I've also added a check that the raw txes are the same: 27c104a

This should ensure that differences in type/0x prefix do not result in differences of tx value before or after property definitition, serialization or hashing

Reply:
Got ya - this makes sense